### PR TITLE
Process review on mozorg string extraction

### PR DESCRIPTION
**Pull request by peiying2**
- Doc content originates from this doc: https://docs.google.com/document/d/1-Pk-bA2tokVB_TSY3VZlbg4Us8IYd3hzu-KHunNOCBg/edit
- Edit Readme doc.

### CODE REVIEW

File: products/mozilla_org/extract_strings.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+You use langcheckers for the following purposes:
+* [app/config/sources.inc.php](https://github.com/mozilla-l10n/langchecker/blob/master/app/config/sources.inc.php): modify this file when there is a new or an updated .lang: add file name, define supported locales, deadline and flags.
+* [app/config/locales.inc.php](https://github.com/mozilla-l10n/langchecker/blob/master/app/config/locales.inc.php): bootstrap a new new locale on Mozilla.org
+
+The following commands are used for some common tasks:
+* `lang_update` is to populate new and revised content in the en-US into localized files.
+* `mark_active` will add an ## active ## tag at the beginning of the file after checking that the entire file is translated and no errors are detected. The associate template will be live on production.
+* `add_tags` tracks the completion of the tagged strings, and add the tag on top of the file when all are translated.
+
+For more details, check [here]()../../tools/webdashboards/langchecker.md).
```
Comment (peiying2) [inline products/mozilla_org/extract_strings.md]: Remove the extra ).

File: products/mozilla_org/README.md [hunk 2]
```diff
@@ -5,5 +5,6 @@ Some documentation specific for common tasks on mozilla.org
 * [Importing translations from existing files](../../tools/webdashboards/import_translations.md).
 * [Updating mozilla.org production](updating_mozillaorg_production.md).
 * [Enable a new locale on production](enable_production.md).
+* [Process overview to expose mozilla.org content for localization](extract_strings.md)
```
Comment (flodolo) [inline products/mozilla_org/README.md]: Should we put this first in the list?

File: products/mozilla_org/extract_strings.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# Process overview to expose mozilla.org content for localization
+
+This document describes the tools and processes involved in converting a mozilla.org template in English to a .lang file containing localizable content and relevant information to help with localization.
+
+## This is a brief overview of the process
+
+* In [Bedrock](https://github.com/mozilla/bedrock), when a page is in the final coding review phase in GitHub and content is finalized, an [L10n label](https://github.com/mozilla/bedrock/pulls?q=is%3Aopen+is%3Apr+label%3AL10N) is added to the PR, signaling that it is ready for string extraction.
+* Strings are extracted from the identified templates, generating a temporary .lang file.
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: A new line like this doesn't have any practical effect in Markdown. You should just continue with `The new strings…` on the same line at this point.

File: products/mozilla_org/extract_strings.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+* Another PR is created in the [Langchecker](https://github.com/mozilla-l10n/langchecker/) repository. Langchecker is the tool used to keep track of all localizable files in the l10n repositories, and contains a set of scripts used to manage the repository.
+* After both PRs are reviewed and approved, a script will be run locally to propagate the en-US content into files in other locales. That means both adding a new file, or adding new strings to existing files.
+* Once merged to the master branch, files will be available on www.mozilla.org repository for localization, and the merge commit will be cherry-picked to [bedrock-l10n](https://github.com/mozilla-l10n/bedrock-l10n), the localization production repository. Within 20 minutes, Pontoon will display the status of the new or updated files for all locales.
+* In some cases, an email through a mailing list will be sent out to communicate the localization request. At this point, the L10n tag is also removed from Bedrock’s PR.
+
+## Environment
+
+In order to run langchecker locally, you need a recent version of PHP (ideally 7 and above):
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: (that's on me) We should use Langchecker consistently in the doc

File: products/mozilla_org/extract_strings.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+It is very important to have a clean, error free .lang file in en-US. Once the file is approved and content is pushed into other locales, if an error is found then, it will take much more effort to correct. If localizers have already started their work, the cleanup is even messier. At that point, it is a fix in one file versus fixes in 100+ files.
+
+Initial review should include checking the file against the page on the testing server and checking against the template. This is to ensure that there are no hard coded strings.
+
+Note: if there is an update to an existing page, it is advised to still generate the .lang file, and copy the new string there instead of copying it from the .html file. This is to make sure, for example, that the syntax used for variable replacements is correct, and remove eventual extra white spaces between words that are not carried over to the .lang file.
+
+### Add notes and comments to the extracted file:
+
+Add important information at the top of the file:
+  `## NOTE: this is a note.`
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: Let's move this into the first bullet point

```
* A Note (`## NOTE:`) for [demo/test URLs](https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/firefox/facebookcontainer/index.lang).
```

File: products/mozilla_org/extract_strings.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+* A Note for [demo/test URLs](https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/firefox/facebookcontainer/index.lang).
+* A Note for test instruction if necessary, for a page that is more dynamic, such as the [privacy_protection_tour.lang](https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/firefox/tracking-protection-tour.lang).
+* Make sure to add tags, if there are strings [bound to l10n tags](working_bedrock.md#l10n-tags).
+
+Proceed to the content of the file. Add a comment to the following type of strings to provide more context:
+* Page title and description. Typically, a page title string should be translated in a short or catchy way
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: Consistency: periods at the end of list items.

File: products/mozilla_org/extract_strings.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+* If the string contains a link, to which [URL](https://github.com/mozilla-l10n/vm_scripts/blob/master/update_mozillaorg.sh0 it points to. This gives context to localizers.
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: Link is broken (check the preview online by clicking View on the file)

File: products/mozilla_org/extract_strings.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+* If a string is behind a tag, make sure to add the [tag binding](https://github.com/mozilla-l10n/langchecker/wiki/.lang-files-format#string-meta-data).
+* A new string behind a tag makes another string obsolete. Make sure to add [an obsolete comment](https://github.com/mozilla-l10n/vm_scripts/blob/master/update_mozillaorg.sh) to help localizers prioritize their work, and that old and new strings are next to each other. This will also make removal of old strings easier in further updates.
+* A comment for an alternative to an expression that may not be culturally relevant, or to a pun that doesn’t translate well.
+* Check English content for capitalization, hyphenated word to make sure they are consistent with the US English style guide (macOS vs. MacOS, sign in to vs. sign into).
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: Add a link to the style guide.

File: products/mozilla_org/extract_strings.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+When all is done, create a PR in the www.mozilla.org repo for review. Describe what the PR is for, add references to the Bedrock’s pull requests to add context for the review.
+
+Check for possible [errors](https://l10n.mozilla-community.org/langchecker/?action=errors) such as duplicates or unmatched strings before requesting for review. In order to do this, you will need to run langchecker locally, and make sure that both the l10n repository and langchecker’s are using the branches you’re working on.
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: Again: Langchecker when it's in the middle of a sentence. The last section is full of them.

File: products/mozilla_org/extract_strings.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+Below are some of the common commands and purposes of using lanchecker.
+
+You use langcheckers for the following purposes:
+* [app/config/sources.inc.php](https://github.com/mozilla-l10n/langchecker/blob/master/app/config/sources.inc.php): modify this file when there is a new or an updated .lang: add file name, define supported locales, deadline and flags.
+* [app/config/locales.inc.php](https://github.com/mozilla-l10n/langchecker/blob/master/app/config/locales.inc.php): bootstrap a new new locale on Mozilla.org
+
+The following commands are used for some common tasks:
+* `lang_update` is to populate new and revised content in the en-US into localized files.
+* `mark_active` will add an ## active ## tag at the beginning of the file after checking that the entire file is translated and no errors are detected. The associate template will be live on production.
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: Wrap `## active ##` with backticks.

File: products/mozilla_org/extract_strings.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+* Make sure to add tags, if there are strings [bound to l10n tags](working_bedrock.md#l10n-tags).
+
+Proceed to the content of the file. Add a comment to the following type of strings to provide more context:
+* Page title and description. Typically, a page title string should be translated in a short or catchy way
+* If the string contains a link, to which [URL](https://github.com/mozilla-l10n/vm_scripts/blob/master/update_mozillaorg.sh0 it points to. This gives context to localizers.
+* If a string is behind a tag, make sure to add the [tag binding](https://github.com/mozilla-l10n/langchecker/wiki/.lang-files-format#string-meta-data).
+* A new string behind a tag makes another string obsolete. Make sure to add [an obsolete comment](https://github.com/mozilla-l10n/vm_scripts/blob/master/update_mozillaorg.sh) to help localizers prioritize their work, and that old and new strings are next to each other. This will also make removal of old strings easier in further updates.
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: Did you want to put a link here? Because the URL is definitely wrong.

**Review by flodolo — COMMENTED** [review]
You forgot to add the document to SUMMARY.

A few nits, but it's pretty much done. If you can fix the remaining issues, we can merge and start asking people to read (and iterate fixes if needed).

File: products/mozilla_org/extract_strings.md (hunk 11 above)
Comment (peiying2) [reply products/mozilla_org/extract_strings.md]: Quite a few of them were wrong. I hope I caught them all.

File: products/mozilla_org/extract_strings.md (hunk 8 above)
Comment (peiying2) [reply products/mozilla_org/extract_strings.md]: I changed the wording a bit, it is the Firefox style guide.

File: products/mozilla_org/extract_strings.md [hunk 12]
```diff
@@ -39,36 +38,36 @@ Note: if there is an update to an existing page, it is advised to still generate
...
-* If the string contains a link, to which [URL](https://github.com/mozilla-l10n/vm_scripts/blob/master/update_mozillaorg.sh0 it points to. This gives context to localizers.
-* If a string is behind a tag, make sure to add the [tag binding](https://github.com/mozilla-l10n/langchecker/wiki/.lang-files-format#string-meta-data).
-* A new string behind a tag makes another string obsolete. Make sure to add [an obsolete comment](https://github.com/mozilla-l10n/vm_scripts/blob/master/update_mozillaorg.sh) to help localizers prioritize their work, and that old and new strings are next to each other. This will also make removal of old strings easier in further updates.
+* Page title and description. Typically, a page title string should be translated in a short or catchy way.
+* If the string contains a link, to which [URL](https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/firefox/tracking-protection-tour.lang#L129) it points to. This gives context to localizers.
+* If a string is behind a tag, make sure to add the [tag binding](https://github.com/mozilla-l10n/langchecker/wiki/.lang-files-format#string-meta-data) it.
```
Comment (flodolo) [inline products/mozilla_org/extract_strings.md]: I think the final *it* should be removed, those are called *tag bindings*.

**Review by flodolo — APPROVED** [review]
I'm a bit afraid of having so many links to .lang file, because they'll get obsolete.

To avoid that, replace `master` with a specific changeset in all of them (the latest is `f5747193ee5af7e83fafa2cf70a2031dfa3d9773`). For example, instead of

```
https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/firefox/tracking-protection-tour.lang#L129
```

do 

```
https://github.com/mozilla-l10n/www.mozilla.org/blob/f5747193ee5af7e83fafa2cf70a2031dfa3d9773/en-US/firefox/tracking-protection-tour.lang#L129
```

Once that's fixed, feel free to merge.

File: SUMMARY.md [hunk 13]
```diff
@@ -20,6 +20,7 @@
 * [Firefox for iOS](products/firefox_iOS/README.md)
     * [Reviewing new strings](products/firefox_iOS/review_iOSstrings.md).
 * [Mozilla.org](products/mozilla_org/README.md)
+    * [Process overview to expose mozilla.org content for localization](extract_strings.md)
```
Comment (flodolo) [inline SUMMARY.md:23]: That's way too long, it should just be "Process overview"

**Pull request merged**